src/main/java/com/moddy/server/service/model/ModelService.java

```diff
@@ -0,0 +1,112 @@
+    private final HairServiceOfferJpaRepository hairServiceOfferJpaRepository;
+    private final PreferOfferConditionJpaRepository preferOfferConditionJpaRepository;
+
+    private Page<HairServiceOffer> findOffers(Long userId, int page, int size){
+        PageRequest pageRequest = PageRequest.of(page-1, size, Sort.by(Sort.Direction.DESC,"id"));
+        Page<HairServiceOffer> offerPage = hairServiceOfferJpaRepository.findByUserId(userId, pageRequest);
+
+        return offerPage;
+    }
+
+    private ModelApplyStatus calModelStatus(boolean apply, boolean offer){
+
+        //테스트 코드에 false true 경우 있는지 테스트하기
+        if(apply == false && offer == false) {
+            return ModelApplyStatus.NOTHING;
+        }
+        else if (apply == true && offer == false) {
+            return ModelApplyStatus.APPLY;
+        }
+        else if (apply == true && offer == true) {
+            return ModelApplyStatus.APPLY_AND_OFFER;
+        }
+        else {
+            throw new NotFoundException(ErrorCode.NOT_FOUND_MODEL_STATUS);
+        }
+
+    }
+
+    public ModelMainResponse getModelMainInfo(Long userId, int page, int size){
+
+        Page<HairServiceOffer> offerPage = findOffers(userId, page, size);
+
+        boolean applyStatus = hairModelApplicationJpaRepository.existsByUserId(userId);
+        boolean offerStatus = hairServiceOfferJpaRepository.existsByUserId(userId);
+        ModelApplyStatus modelApplyStatus = calModelStatus(applyStatus, offerStatus);
+        User user = modelJpaRepository.findById(userId).orElseThrow(() -> new NotFoundException(ErrorCode.NOT_FOUND_MODEL_INFO));
+
+        if(modelApplyStatus != ModelApplyStatus.APPLY_AND_OFFER){
+            return new ModelMainResponse(
+                    page,
+                    size,
+                    modelApplyStatus,
+                    user.getName(),
+                    new ArrayList<>()
+            );
+        }
+
+        List<HairServiceOffer> hairServiceOffers = offerPage.stream().toList();
+        List<OfferResponse> offerResponseList = new ArrayList<>();
+
+        for (int i = 0; i < hairServiceOffers.size(); i++){
+            Designer designer = hairServiceOffers.get(i).getDesigner();
+            List<PreferOfferCondition> preferOfferCondition = preferOfferConditionJpaRepository.findTop2ByHairServiceOfferId(hairServiceOffers.get(i).getId());
+            List<OfferCondition> offerConditionTop2List = preferOfferCondition.stream().map(PreferOfferCondition::getOfferCondition).collect(Collectors.toList());
+
+            OfferResponse offerResponse = new OfferResponse(
+                    hairServiceOffers.get(i).getId(),
+                    designer.getProfileImgUrl(),
+                    designer.getName(),
+                    designer.getHairShop().getName(),
+                    offerConditionTop2List,
+                    hairServiceOffers.get(i).getIsClicked());
+
+            offerResponseList.add(offerResponse);
+        }
```

Comment:
```suggestion
        List<OfferResponse> hairServiceOffers = offerPage.stream().map(offer -> {
            Designer designer = offer.getDesigner();
            List<PreferOfferCondition> preferOfferCondition = preferOfferConditionJpaRepository.findTop2ByHairServiceOfferId(offer.getId());
            List<OfferCondition> offerConditionTop2List = preferOfferCondition.stream().map(PreferOfferCondition::getOfferCondition).collect(Collectors.toList());
            OfferResponse offerResponse = new OfferResponse(
                    hairServiceOffers.get(i).getId(),
                    designer.getProfileImgUrl(),
                    designer.getName(),
                    designer.getHairShop().getName(),
                    offerConditionTop2List,
                    hairServiceOffers.get(i).getIsClicked());
        })
```
p2
stream을 사용하여 이런 식으로 코드를 줄일 수 있을 것 같습니다.

Reply:
```java
List<OfferResponse> offerResponseList = offerPage.stream().map(offer -> {
            Designer designer = offer.getDesigner();
            List<PreferOfferCondition> preferOfferCondition = preferOfferConditionJpaRepository.findTop2ByHairServiceOfferId(offer.getId());
            List<OfferCondition> offerConditionTop2List = preferOfferCondition.stream().map(PreferOfferCondition::getOfferCondition).collect(Collectors.toList());

            OfferResponse offerResponse = new OfferResponse(
                    offer.getId(),
                    designer.getProfileImgUrl(),
                    designer.getName(),
                    designer.getHairShop().getName(),
                    offerConditionTop2List,
                    offer.getIsClicked()
            );
            return offerResponse;
        }).collect(Collectors.toList());
```

요런식으로 구현했습니다~~!